### DISCUSSION TITLE
How to write Markdown in a predominantly right-to-left language (Hebrew), interspersed with words and code blocks in a left-to-right language (English)?

### DISCUSSION THREAD

**Discussion by ItaiBEdu** [community/community#65545] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I'd like to write a README.md file in Hebrew interspersed with some English words and code blocks . As long as the language is solely Hebrew, or if Hebrew and English are kept in separate paragraphs, GitHub renders the Hebrew text correctly, namely the paragraphs are right-aligned, and the words flow from right to left, as can be seen in the following paragraph.

אם נניח כדור במקום כלשהו בתוך קערה, הוא יתגלגל מטה וייעצר בסופו של דבר בתחתיתה. זה קורה כי הכדור שואף להגיע לנקודה שבה יש לו הכי מעט אנרגיה – כלומר למטה. לכן, כל עוד הוא נמצא על משטח משופע, הוא ינצל את האפשרות להתגלגל לנקודה נמוכה יותר. בסופו של דבר הוא ייעצר רק כשיגיע למשטח אופקי, כלומר לתחתית הקערה, או בשפה קצת יותר מתמטית: לנקודת המינימום של האנרגיה. זהו שיווי משקל יציב – הכדור יישאר בו לאורך זמן, וגם אם נזיז אותו הצידה הוא יתגלגל מחדש לאותה נקודת מינימום.

However, if a Hebrew paragraph starts with an English word, it is rendered incorrectly: the paragraph is left-aligned, and the flow of text between English phrases and Hebrew phrases is left-to-right:

"Does anyone speak English?" שאל התייר.

Another problem occurs when an English phrase that ends with a punctuation mark is embedded inside a Hebrew paragraph. The punctuation mark is positioned wrongly:

אנא הדפיסו את המשפט "Hello, world!"

Another point of failure occurs when a code block (in formalized English) is embedded inside a Hebrew list:

1. צעד ראשון: העתיקו את הקוד הבא.
   ```java
   public static void main(String ... args)
   {
       System.out.println(???);
   }
   ```

One good workaround that is available, for instance, in Replit, is to use raw HTML rather than MarkDown, and wrap the entire text in `<div align="right", dir="rtl", lang="he"> ... </div>`. Syntax-highlighted codeblocks can then be added with `<pre><code class="language-java"> ... </code></pre>`. However, according to the answers I received to [this question](https://github.com/orgs/community/discussions/65381#discussion-5572811), this method doesn't work in GitHub.

Is there a viable solution for the problem of writing a README.md file in a right-to-left language (not only Hebrew, but also Arabic and Farsi; altogether the native tongues of hundreds of millions of people)? Note that embedding screenshots of code and of English phrases is not a viable solution, as it is unwieldy, and opaque to edits, analysis, and copy-and-paste.

I should also point out that text editors with (correct) bidirectional support exist for decades now. That's "decade" with a plural "s". This is not a problem that requires a fresh solution.

EDIT: For future reference, I have reported some right-to-left problems with the current GitHub rendering engine in [this reply](https://github.com/orgs/community/discussions/8115#discussioncomment-6872844), which was posted in the discussion that is linked in [this announcement](https://github.blog/changelog/2022-01-31-editor-support-for-right-to-left-languages/).

**Comment by danieldia-dev** [comment]
I had a similar issue with Arabic, and to be honest I still have no clue how to solve it. My only solution was to spend a lot of time editing every damn pixel (hyperbola) to make sure the Arabic text looked alright. To be fair, for some stupid reason using a phone was less problematic than typing through a PC (I have no clue why, but I suspect it's the autocorrect stuff that makes it more bearable)...

**Reply by ItaiBEdu** [reply]
Thanks. This is clearly not an acceptable solution. If speakers of left-to-right languages had to edit pixels in order for their README.md files to render correctly, I doubt GitHub would even exist today.

**Comment by danieldia-dev** [comment]
I found you something: [(https://stackoverflow.com/questions/2258028/rtl-in-markdown)]
This hould probably do it, and good luck!

**Reply by ItaiBEdu** [reply]
Thanks. However, as I mentioned in my original post, it seems [GitHub doesn't support syntax-highlighted codeblocks in raw HTML](https://github.com/orgs/community/discussions/65381#discussion-5572811).

I'm a teacher. Suppose I want to write a README.md file for my students with step-wise instructions, such as:

1. Copy the following class stub.
    ```java
    public class HelloWorld
    {
        public static void main(String ... args)
        {
            System.out.println("Hello, world!")
        }
    }
    ```
2. Change the code so it writes "Goodbye, universe?" instead of "Hello, world!".

However, the instructions should be in Hebrew (or Arabic, for that matter). How can I do this?
